### ISSUE TITLE
In what part of the code ContrastiveLoss is used?

### ISSUE THREAD

**Issue by Rezamh13** [PRBonn/segcontrast#18]
Hi, I was browsing through the code, in `contrastive_train.py` where you imported `ContrastiveLoss` but set `criterion = nn.CrossEntropyLoss().cuda()`. Could you explain where exactly `ContrastiveLoss` is used?

**Comment by nuneslu** [comment]
Hi, this is because we used the contrastive loss formulation from [MoCo](https://github.com/facebookresearch/moco) paper. They "simplify" the loss to a cross-entropy function. In the end is the same, but it simplifies the similarity computation between positives and negatives.

**Comment by Rezamh13** [comment]
@nuneslu Could you share the pre-training loss curve?

**Comment by nuneslu** [comment]
Hi, yes, this is the loss curve:
![image](https://user-images.githubusercontent.com/6524770/225581368-270a27c7-b4f4-41b8-95fd-a378868a8b20.png)

But note that in the beginning is normal for the loss to increase a bit since it is still accumulating segments in the feature bank: 
![image](https://user-images.githubusercontent.com/6524770/225581580-4c46f67d-63b8-42e1-95f9-fa3e97465e42.png)

**Comment by nuneslu** [comment]
Also, in case you are interested, we have just released the code for our new pre-training method here https://github.com/PRBonn/TARL

**Comment by nuneslu** [comment]
I will close this now. If you have any more questions, feel free to reopen it.
